### PR TITLE
Add to this repo the same coverage checks as we have in core repo (#261)

### PR DESCRIPTION
**Pull request by hanjm**
**Description:**
add .codecov.yml, add badge to README.md

**Link to tracking Issue:** 
#261

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+coverage:
+  precision: 2
+  round: down
+  range: "80...100"
+  status:
+    project:
+      default:
+        enabled: yes
+        target: 85%  # This needs to be updated to 90 or 95.
```
Comment (bogdandrutu) [inline .codecov.yml:14]: We need to lower this to maybe 78 until we hit the target of 85 :)

**Review by bogdandrutu — APPROVED** [review]

File: .codecov.yml (hunk 1 above)
Comment (bogdandrutu) [reply .codecov.yml:14]: Maybe 79 :)

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  range: "80...100"
+  status:
+    project:
+      default:
+        enabled: yes
+        target: 79%  # This needs to be updated to 85 or higher.
+    patch:
+      default:
+        enabled: yes
+        target: 95%
```
Comment (tigrannajaryan) [inline .codecov.yml:18]: Can we set lower target for contrib?

Comment (bogdandrutu) [reply .codecov.yml:18]: Let's see what happens? :)

**Review by bogdandrutu — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@bogdandrutu Merging, but I suspect we may need to relax the requirements a bit for this repo :-)

**Pull request merged**